src/main/java/org/sitenv/ccdaparsing/processing/MedicalEquipmentProcessor.java

```diff
@@ -0,0 +1,137 @@
+package org.sitenv.ccdaparsing.processing;
+
+import org.apache.log4j.Logger;
+import org.sitenv.ccdaparsing.model.CCDAID;
+import org.sitenv.ccdaparsing.model.CCDAMedicalEquipment;
+import org.sitenv.ccdaparsing.model.CCDAMedicalEquipmentOrg;
+import org.sitenv.ccdaparsing.model.CCDANonMedicalSupplyAct;
+import org.sitenv.ccdaparsing.util.ApplicationConstants;
+import org.sitenv.ccdaparsing.util.ApplicationUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.AsyncResult;
+import org.springframework.stereotype.Service;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.xml.transform.TransformerException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Future;
+
+@Service
+public class MedicalEquipmentProcessor {
+    private static final Logger logger = Logger.getLogger(MedicalEquipmentProcessor.class);
+
+    @Autowired
+    ProcedureProcessor procedureProcessor;
+
+    @Async()
+    public Future<CCDAMedicalEquipment> retrieveMedicalEquipments(XPath xPath , Document doc) throws XPathExpressionException, TransformerException {
```

Comment:
A few more examples of 'equipments' in this class which I think should be singular. I don't want to be pedantic, so, if you don't agree, don't worry about updating.